internalservice_agentmanager.go

```diff
@@ -0,0 +1,331 @@
+package gocbcorex
+
+import (
+	"context"
+	"crypto/tls"
+	"errors"
+	"net/http"
+	"sync"
+
+	"golang.org/x/exp/slices"
+
+	"go.uber.org/zap"
+)
+
+type InternalServiceAgentManagerReconfigureOptions struct {
+	TLSConfig     *tls.Config
+	Authenticator Authenticator
+}
+
+type InternalServiceAgentManagerOptions struct {
+	Logger *zap.Logger
+
+	TLSConfig          *tls.Config
+	Authenticator      Authenticator
+	SeedConfig         SeedConfig
+	CompressionConfig  CompressionConfig
+	ConfigPollerConfig ConfigPollerConfig
+	HTTPConfig         HTTPConfig
+}
+
+type internalServiceAgentManagerState struct {
+	tlsConfig     *tls.Config
+	authenticator Authenticator
+	httpTransport *http.Transport
+
+	latestConfig *ParsedConfig
+}
+
+type InternalServiceAgentManager struct {
+	bucketsLock sync.Mutex
+	topoLock    sync.Mutex
+	opts        InternalServiceAgentManagerOptions
+	logger      *zap.Logger
+	userAgent   string
+	networkType string
+
+	state *internalServiceAgentManagerState
+
+	clusterAgent *Agent
+	bucketAgents map[string]*Agent
+	closed       bool
+
+	bucketsWatcher  *StreamWatcherHttp[[]bucketDescriptor]
+	watchersCancel  func()
+	needsBucketChan chan struct{}
+
+	topologyCfgWatcher *ConfigWatcherHttp
+}
+
+type bucketDescriptor struct {
+	Name string
+	UUID string
+}
+
+func CreateInternalServiceAgentManager(ctx context.Context, opts InternalServiceAgentManagerOptions) (*InternalServiceAgentManager, error) {
```

Comment:
We should probably name it based on it's behavior rather than our presumptions about its use.  InternalServiceAgentManager implies its only useful for internal services, naming it something like "AllBucketsAgentManager" or... "WatchingAgentManager" is probably more descriptive (I dislike those names though, there _must_ be something better).